test/powershell/Modules/Microsoft.PowerShell.Management/Get-PSProvider.Tests.ps1

```diff
@@ -24,4 +24,56 @@ Describe "Get-PSProvider" -Tags "CI" {
 
 	{ $actual | Format-List } | Should -Not -Throw
     }
+
+    Context 'PathSeparators' {
+        BeforeAll {
+            $windowsTestCases = @(
+                @{Provider = 'FileSystem'; Value = @("\", "/")}
+                @{Provider = 'Variable'; Value = @("\", "/")}
+                @{Provider = 'Function'; Value = @("\", "/")}
+                @{Provider = 'Registry'; Value = @("\")}
+            )
+
+            $nonWindowsTestCases = @(
+                @{Provider = 'FileSystem'; Value = @("/", "\")}
+                @{Provider = 'Variable'; Value = @("/", "\")}
+                @{Provider = 'Function'; Value = @("/", "\")}
+            )
+
+            $testCases = if ($IsWindows) {
+                            $windowsTestCases
+                        }
+                        else {
+                            $nonWindowsTestCases
+                        }
+        }
+
+        It '<Provider> provider has PathSeparator property' -TestCases $testCases {
+            param ($Provider, $Value)
+
+            (Get-PSProvider $Provider).PathSeparator | Should -Be $Value
+        }
+
+        It 'cannot modify PathSeparator collection in <Provider> provider' -TestCases $testCases {
+            param ($Provider, $Value)
+
+            $separator = (Get-PSProvider $Provider).PathSeparator
+
+            { $separator[0] = "w" } | Should -Throw
```

Comment:
Maybe add one test to check that the public `PathSeparator` is get-only

Reply:
Done